signer/hsm.go

```diff
@@ -107,6 +107,34 @@ type GCPHSM struct {
 	GenericHSM
 }
 
+type UnlimitedBytesRandReader struct {
+	rr crypto11.PKCS11RandReader
+}
+
+func (ubrr *UnlimitedBytesRandReader) Read(data []byte) (n int, err error) {
+	var result []byte
+	need := len(data)
+	for need > 0 {
+		next := make([]byte, 1024)
```

Comment:
`min(need, 1024)` or similar?

Reply:
Yeah, this needs some more iteration (and tests)!